src/haystack/filter/lexer.rs

```diff
@@ -253,10 +254,11 @@ impl<'a, R: Read> Lexer<Scanner<'a, R>> {
                 if !self.scanner.is_lower() {
                     return self.scanner.make_generic_err("Expecting path segment");
                 }
-            } else if !self.scanner.is_lower() {
+            } else {
```

Comment:
Here's the fix. Once `parse_path` has been entered, the first part of the path (i.e. siteRef->) has been tokenized. From here on in, we're iterating for more `->` paths. Hence I've just changed the conditional that if `->` is not found next then it just exits for this token.